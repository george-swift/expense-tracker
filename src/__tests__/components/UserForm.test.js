import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Form from '../../components/UserForm';

let loggedIn = false;
const username = 'Test Dev';
const email = 'testdev@email.com';
const password = 'testing101';
const passwordConfirmation = 'testing101';
const handleChange = jest.fn(() => {});
const submitAction = jest.fn(() => {});
const switchAction = jest.fn(() => {});

describe('Snapshot of the user form component', () => {
  it('should render correctly on sign up', () => {
    const tree = renderer
      .create(
        <Router>
          <Form
            loggedIn={loggedIn}
            username={username}
            email={email}
            password={password}
            passwordConfirmation={passwordConfirmation}
            handleChange={handleChange}
            submitAction={submitAction}
            switchAction={switchAction}
          />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render edit form correctly if logged in', () => {
    loggedIn = true;
    const tree = renderer
      .create(
        <Router>
          <Form
            loggedIn={loggedIn}
            username={username}
            email={email}
            handleChange={handleChange}
            submitAction={submitAction}
          />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
