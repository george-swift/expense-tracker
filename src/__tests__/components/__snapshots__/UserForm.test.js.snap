// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of the user form component should render correctly on sign up 1`] = `
<form
  className={null}
  onSubmit={[MockFunction]}
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Username"
    >
      Username
    </label>
    <input
      className="form-control"
      minLength={2}
      name="username"
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="Test Dev"
    />
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Email"
    >
      Email
    </label>
    <input
      className="form-control"
      name="email"
      onChange={[MockFunction]}
      required={true}
      type="email"
      value="testdev@email.com"
    />
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Password"
    >
      Password
    </label>
    <input
      className="form-control"
      minLength={6}
      name="password"
      onChange={[MockFunction]}
      required={true}
      type="password"
      value="testing101"
    />
    <small
      className="fst-italic text-danger d-none"
    >
      Password and password confirmation do not match. Try again
    </small>
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Password Confirmation"
    >
      Password Confirmation
    </label>
    <input
      className="form-control"
      minLength={6}
      name="password_confirmation"
      onChange={[MockFunction]}
      required={true}
      type="password"
      value="testing101"
    />
    <small
      className="fst-italic text-danger d-none"
    >
      Password and password confirmations do not match. Try again
    </small>
  </div>
  <div>
    <button
      className="btn mb-3 w-100"
      type="submit"
    >
      Sign up
    </button>
    <p
      className="text-center"
    >
      <span
        className="fst-italic"
      >
        Already have an account?
      </span>
       
      <a
        href="/login"
        onClick={[Function]}
      >
        Click here to log in
      </a>
    </p>
  </div>
</form>
`;

exports[`Snapshot of the user form component should render edit form correctly if logged in 1`] = `
<form
  className="py-3"
  onSubmit={[MockFunction]}
>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Username"
    >
      Username
    </label>
    <input
      className="form-control"
      minLength={2}
      name="username"
      onChange={[MockFunction]}
      required={true}
      type="text"
      value="Test Dev"
    />
  </div>
  <div
    className="mb-3"
  >
    <label
      className="form-label"
      htmlFor="Email"
    >
      Email
    </label>
    <input
      className="form-control"
      name="email"
      onChange={[MockFunction]}
      required={true}
      type="email"
      value="testdev@email.com"
    />
  </div>
  <div
    className="mb-3"
  >
    <button
      className="btn w-100"
      type="submit"
    >
      Update Profile
    </button>
  </div>
</form>
`;
